ArjunaJTA/jdbc/classes/com/arjuna/ats/internal/jdbc/recovery/BasicXARecovery.java

```diff
@@ -202,7 +204,9 @@ private final JDBC2RecoveryConnection getConnection ()
 			if (dc != null)
 				dbProperties.put(TransactionalDriver.dynamicClass, dc);
 
-			return new JDBC2RecoveryConnection(theURL, dbProperties);
+			JDBC2RecoveryConnection connection = new JDBC2RecoveryConnection(theURL, dbProperties);
+			connections.add(connection);
```

Comment:
They would be held for the lifetime of the JVM but that would actually be OK because we may want to rescan the XAResource periodically.  That being said, for the time being I don't think it would be rescanned as hasMoreResources does not reset - I can fix that